### ISSUE TITLE
Handling Batch normalization in Horovod in TensorFlow

### ISSUE THREAD

**Issue by calledbymountains** [uber/horovod#270]
What is the correct way to handle batch normalization when using horovod in TensorFlow ?

In the standard case I suppose, it will also aggregate the statistics of each batch normalization layer in each  GPU, which I think is wrong.

**Comment by alsrgv** [comment]
@calledbymountains, Horovod handles each worker independently from Batch Norm perspective, so if you run 1 GPU per process, each GPU will have its own BN statistics.  See #49 for more discussion.

**Comment by calledbymountains** [comment]
@alsrgv  
Okay. I quote you from #49 , where you say that " I believe if you add batch normalization to the train_op following example here, the batch norm would run on every worker independently, without exchanging or averaging those updates. Do you anticipate that creating any problems? " 

Now, if something is added to the train_op, then does it mean that horovod would only exchange or average only the gradients of those operations ? As far as I know, batch normalization does not introduce gradients. This is just for future clarity. 
And, with respect to #49, would that not be a problem when trying to do batch renormalization ?

**Comment by alsrgv** [comment]
Basically, with enough uniformity in the way data is processed by workers, BN statistics on each worker should be approximately the same - and so BN and renormalization should work as expected.

The list of gradients averaged by Horovod is driven by the gradients returned from `optimizer.compute_gradients()` - which normally computes all gradients w.r.t. loss.

**Comment by calledbymountains** [comment]
Okay. I understand the point.
